src/backends/matrix_state_interleaved.hpp

```diff
@@ -0,0 +1,248 @@
+#pragma once
+
+#include <memory/memory.hpp>
+#include <util/span.hpp>
+#include <util/rangeutil.hpp>
+
+#include "gpu_kernels.hpp"
+
+namespace nest {
+namespace mc {
+namespace gpu {
+
+/// matrix state
```

Comment:
Commenting style: here and below is a mix of blank lines for spacing, blank comments for spacing, `///` and `//`, number of spaces before comment text, and comments as phrases and sentences. This is messy and should be made more consistent.